### PR TITLE
Polynomial Multiplication Using Linked Lists

### LINKED ISSUE #1546: [Idea]: <Polynomial Multiplication Using Linked Lists>
### Idea Title

Polynomial Multiplication Using Linked Lists

### Idea Description

**Polynomial multiplication using singly linked lists**

Write a C++ program to multiply two polynomials using singly linked lists. Each polynomial should be represented as a linked list where each node contains a coefficient and an exponent for a term. The program should take user input for two polynomials, perform the multiplication, and display the result.

**Polynomial Representation:**
A polynomial is represented using a singly linked list. Each node in the list contains an integer coefficient for the coefficient of the term and an integer exponent for the exponent of the term. A pointer links the next node in the list.

**Tasks:**
1) Implement a function to multiply two polynomials (stored as linked lists).
2) Store the result in a new linked list, combining like terms.

**Input:**
1. First polynomial: number of terms, followed by coefficient and exponent pairs.
2. Second polynomial: number of terms, followed by coefficient and exponent pairs.

**Output:**
1. Print the first and second polynomials in standard format.
2. Print the resulting polynomial after multiplication, simplified.
3. If both polynomials are zero, print "Both polynomials are empty."

### Potential Benefits

- [X] Improves user experience
- [X] Enhances learning
- [X] Adds a new feature
- [x] Improves code quality
- [ ] Other

### Implementation Suggestions (Optional)

_No response_

### PR DESCRIPTION
**Pull request by nishthaa-m**
### Description
This pull request introduces a new feature that implements **Polynomial Multiplication using a linked list data structure**. The implementation allows users to define polynomials by specifying terms with coefficients and exponents, and then multiply two polynomials together, returning the resulting polynomial. This feature enhances the functionality of polynomial operations in the project.

The motivation behind this change is to provide a flexible and efficient way to handle polynomial calculations, making it easier for users to work with mathematical expressions programmatically.

Fixes #1546

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

**Test Cases:**

image
image

### CODE REVIEW

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
we need only .md file so, can you write your code in markdown file for our docs

**Comment by nishthaa-m** [comment]
@ajay-dhangar Made the necessary changes.
Kindly review them.

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
We need only `.md` files and they must be in the `docs` folder. Please remove any other files, such as `.cpp` files.

**Comment by nishthaa-m** [comment]
@ajay-dhangar Updated it

File: docs/linked-list/Polynomial_multiplication.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    // Display result
+    cout << "Result of multiplication: ";
+    if (result) {
+        display(result);
+    } else {
+        cout << "Both polynomials are empty." << endl;
+    }
+
+    return 0;
+}
```
Comment (ajay-dhangar) [inline docs/linked-list/Polynomial_multiplication.md:154]: close your code example with ```` ``` ````

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**